.github/workflows/docker.yml

```
@@ -0,0 +1,75 @@
+name: docker
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build-and-push:
+    name: build-and-push
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Determine tag name
+        id: determine_tag
+        shell: bash
+        run: |
+          if [[ -z $GITHUB_BASE_REF ]]; then
```

```suggestion
          if [[ -z "${GITHUB_BASE_REF}" ]]; then
```
To be on the safe side, although `[[` expressions also work without quotes `"`.